src/addons/withSoundCloudAudio.js

```diff
@@ -52,6 +53,32 @@ export default function withSoundCloudAudio (WrappedComponent) {
       this.listenAudioEvents();
     }
 
+    componentDidUpdate(prevProps) {
+      if (this.props.streamUrl === prevProps.streamUrl) return;
+
+      if (this.props.streamUrl) {
+        this.setState({ hasSetStartTime: false }, () => {
+          this.requestAudio();
+
+          !prevProps.streamUrl && this.listenAudioEvents();
+
+          this.state.playing && this.soundCloudAudio.play();
+        });
+      } else {
+        this.soundCloudAudio.stop();
+
```

Comment:
I wasn't sure if `resetPlayedStore();` should be called here also, I left it out currently.